app/src/main/kotlin/com/mapzen/erasermap/view/DynamicChildHeightRelativeLayout.kt

```diff
@@ -0,0 +1,44 @@
+package com.mapzen.erasermap.view
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.View
+import android.widget.RelativeLayout
+import java.util.ArrayList
+
+abstract class DynamicChildHeightRelativeLayout : RelativeLayout {
+
+  constructor(context: Context) : super(context) {}
+  constructor(context: Context, attributes: AttributeSet) : super(context, attributes) {}
+  constructor(context: Context, attributes: AttributeSet, style: Int) : super(context, attributes,
+      style) {}
+
+  abstract fun idsForDynamicChildren(): ArrayList<Int>
+
+  var views: ArrayList<View>? = null
+
+  override fun onFinishInflate() {
+    super.onFinishInflate()
+    views = ArrayList<View>()
+    for (i in 0..idsForDynamicChildren().size - 1) {
+      val viewId = idsForDynamicChildren()[i]
+      (views as ArrayList<View>).add(findViewById(viewId))
+    }
+  }
+
+  override fun onLayout(changed: Boolean, l: Int, t: Int, r: Int, b: Int) {
+    super.onLayout(changed, l, t, r, b)
```

Comment:
Calling `super.onLayout(...)` here will cause the dynamic children to be laid out twice which is not super performant. Perhaps we could avoid with something like this-- borrowed from `RelaiveLayout.onLayout(...)`.

```kotlin
  override fun onLayout(changed: Boolean, l: Int, t: Int, r: Int, b: Int) {
    val dynamicViews = views
    if (dynamicViews == null) {
      super.onLayout(changed, l, t, r, b)
      return
    }

    val count = childCount
    val height = Math.abs(b - t)

    for (i in 0..count - 1) {
      val child = getChildAt(i)

      if (child.visibility != GONE) {          
        val params = child.layoutParams as RelativeLayout.LayoutParams
        if (child in dynamicViews) {
          layoutView(child, b + height)
        } else {
          child.layout(params.mLeft, params.mRight, params.mTop, params.mBottom)
        }
      }
    }
  }
```

The only catch here is that `mLeft`, `mRight`, `mTop`, `mBottom` are private members of `RelativeLayout.LayoutParams` 😞  but maybe we can find a way to workaround that. If not I think its fine to keep as is for now and just keep an eye out for jank when scrolling the list.